documentation/tutorial.md

```diff
@@ -160,93 +168,77 @@ selectors in Crawly shell.
 
 ```
-{:ok, document} = Floki.parse_document(response.body)
-document |> Floki.find("section.wrapper") |> Floki.find("div.article-tiles.article-tiles--wide a") |> Floki.attribute("href")
+response = Crawly.fetch("https://books.toscrape.com/")
 
-["/our-range/tools/power-tools", "/our-range/tools/garage-storage",
- "/our-range/tools/hand-tools", "/our-range/tools/tool-storage",
- "/our-range/tools/ladders", "/our-range/tools/safety-equipment-and-workwear",
- "/our-range/tools/work-benches"]
 ```
 
-The result of running the command above is a list of elements which
-contain href attribute of links selected with
-`a.category-block-heading__title` CSS selector. These URLs will be
-used in order to feed Crawly with requests to follow.
-
-In order to find the proper CSS selectors to use, you might find
-useful opening the target page from the shell in your web browser. You
-can use your browser developer tools to inspect the HTML and come up
-with a selector.
-
-Now let's navigate to one of the Homebase product pages and extract
-data from it.
+All books on the page are enclosed in a `.product_pod` class. We can the select the class and pick the information we want from it.
 
 ```
-response = Crawly.fetch("https://www.homebase.co.uk/bosch-universalimpact-800-corded-hammer-drill_p494894")
-
+items =
+  document
+  |> Floki.find(".product_pod")
+  |> Enum.map(fn x ->
+    %{
+      title: Floki.find(x, "h3 a") |> Floki.attribute("title") |> Floki.text(),
+      price: Floki.find(x, ".product_price .price_color") |> Floki.text()
+    }
+  end)
 ```
 
-Extract the `title` with:
+Here we select all products using the class name, iterate through it and pick the title using `Floki.find(x, "h3 a") |> Floki.attribute("title") |> Floki.text()`
+and pick the price using `Floki.find(x, ".product_price .price_color") |> Floki.text()`. The full text of the book name is only contained
+in the title attribute and not in the text node of the `a` tag, hence our use of the title attribute.
 
-```
-{:ok, document} = Floki.parse_document(response.body)
-document |> Floki.find(".page-title h1") |> Floki.text()
-"Bosch UniversalImpact 800 Corded Hammer Drill"
-```
-
-Extract the `SKU` with:
+To handle pagination, we'll tell crawly to visit the next page by extracting the next pages' urls and building new requests from them.
 
 ```
-document |> Floki.find(".product-header-heading span") |> Floki.text
-"SKU:  494894"
+next_requests =
+  document
+  |> Floki.find(".next a")
+  |> Floki.attribute("href")
+  |> Enum.map(fn url ->
+    Crawly.Utils.build_absolute_url(url, response.request.url)
+    |> Crawly.Utils.request_from_url()
+  end)
 ```
 
-Extract the `price` with:
-
-```
-document |> Floki.find(".price-value [itemprop=priceCurrency]") |> Floki.text
-"£82"
-```
+Here we get the url for the next page using Floki, which could be an empty string if such doesn't exist, this would mean that we have gotten to the last page and will result in the spider stopping. 
```

Comment:
would it really result in an empty string? `Floki.attribute` should return an empty list if it does not find any nodes with `href`s

Reply:
Done